src/ArrayCollection.php

```diff
@@ -98,7 +104,7 @@ public function first()
      * @psalm-template K of array-key
      * @psalm-template V
      */
-    protected function createFrom(array $elements)
+    protected function createFrom(array $elements): static
```

Comment:
Adding a native return type is a breaking change. We must not do that in a minor or bugfix release. Apart from that, this change seems unrelated to the goal of this PR.